component/src/action_handler/actions/swap_claim.rs

```diff
@@ -50,7 +39,7 @@ impl ActionHandler for SwapClaim {
         // 1. Validate the epoch duration passed in the swap claim matches
         // what we know.
         let epoch_duration = state.get_epoch_duration().await?;
-        let provided_epoch_duration = swap_claim.body.epoch_duration;
+        let provided_epoch_duration = swap_claim.epoch_duration;
```

Comment:
I moved the `epoch_duration` out of the body of the swap claim, because it should be considered as a public input to the proof which the user happens to supply in the transaction, like the commitment tree anchor, and not something that should be signed over.